src/monster.cpp

```diff
@@ -1059,22 +1059,21 @@ bool monster::can_act() const
              ( !has_effect( effect_stunned ) && !has_effect( effect_downed ) && !has_effect( effect_webbed ) ) );
 }
 
-int monster::sight_range( const int light_level ) const
+int monster::sight_range( const float light_level ) const
 {
     // Non-aquatic monsters can't see much when submerged
     if( !can_see() || effect_cache[VISION_IMPAIRED] ||
         ( underwater && !swims() && !has_flag( MF_AQUATIC ) && !digging() ) ) {
         return 1;
     }
-    static const int default_daylight = default_daylight_level();
+    static const float default_daylight = default_daylight_level();
     if( light_level == 0 ) {
         return type->vision_night;
-    } else if( light_level == default_daylight ) {
+    } else if( light_level >= default_daylight ) {
```

Comment:
Was this perhaps intended as some sort of blinding? Because it seems reasonable to me that a floodlight could reduce the sight range of someone in its area.
As an aside, there is a linear_interpolation function [here](https://github.com/CleverRaven/Cataclysm-DDA/blob/20872a7e7900a080c43b6659a261f1859fba86f3/src/suffer.cpp#L873-L886). Maybe that could be used here to make the intent clearer (and maybe moved to some math/helper functions file?). Something like
```cpp
if( light_level <= default_daylight) {
  return linear_interpolation( 0, type->vision_night, 
                default_daylight, type->vision_day );
} else {
  return linear_interpolation( default_daylight, type->vision_day,
                             2*default_daylight, type->vision_night);
}
```
Then, at twice the usual brightness you can't see further than your nightvision as you must shield your eyes from the light. Just a thought.

Reply:
I think cases of illumination level over 100 are bending system to beyond its limitations and should generally be ignored.

Reply:
Okay, I'm currently looking through the code but can't really find exact definitions of how brightness should be handled. For example, [default_daylight_level](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/calendar.cpp#L47) is still a double. Are values above 100 impossible? I'm thinking that refactoring light level into its own class would be nice to ensure some invariants. I may have a go at it myself once this PR is merged.

Reply:
I've decided to take a look at moving the light level into its own class. Once that is well encapsulated, it should just be a matter of defining some constants and modifying the addition etc. to respect any linear/non-linear scale, upper bounds etc.